### PR TITLE
Add new Grid mode for NGE Instantiate on Volume

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15180/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15180/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15180/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15180/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15180/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15180/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15180/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15180/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15180/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15180/merge/testResults/webgpuplaywright/index.html

**Review by RaananW — APPROVED** [review]

File: packages/dev/core/src/Meshes/Node/Blocks/Instances/instantiateOnVolumeBlock.ts [hunk 1]
```diff
@@ -168,12 +191,58 @@ export class InstantiateOnVolumeBlock extends NodeGeometryBlock implements INode
             const boundingInfo = extractMinAndMax(this._vertexData.positions!, 0, this._vertexData.positions!.length / 3);
             const min = boundingInfo.minimum;
             const max = boundingInfo.maximum;
-            const direction = new Vector3(1, 0, 0);
+            const direction = new Vector3(0.5, 0.8, 0.2);
```
[the remark this answers is not in the document]
Comment (deltakosh) [reply packages/dev/core/src/Meshes/Node/Blocks/Instances/instantiateOnVolumeBlock.ts:194]: random :)

**Pull request merged**